rpc/call_raw.go

```diff
@@ -111,11 +113,28 @@ func (c *Client) callBatchMethods(ctx context.Context, msgs json.RawMessage) str
 // callSingleMethod executes single JSON-RPC message and constructs proper response.
 func (c *Client) callSingleMethod(ctx context.Context, msg json.RawMessage) string {
 	// unmarshal JSON body into json-rpc request
-	chainID, method, params, id, err := methodAndParamsFromBody(msg)
+	jsonrpcParams, err := methodAndParamsFromBody(msg)
 	if err != nil {
-		return newErrorResponse(errInvalidMessageCode, err, id)
+		return newErrorResponse(errInvalidMessageCode, err, nil)
+	}
+
+	chainID := jsonrpcParams.ChainID
+	method := jsonrpcParams.Method
+	params := jsonrpcParams.Params
+	timeout := jsonrpcParams.Timeout
```

Comment:
Since we're adding timeout support, I wonder if it would be useful to have a high default timeout to avoid something hanging for too long or forever. Or maybe that's not a possibility at all?

Reply:
we could default it, but I'd rather keep the behavior as it is for now, to avoid introducing bugs, if that's ok